api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -487,28 +488,20 @@ public Relationship saveRelationship(Relationship relationship) throws APIExcept
 	 *      java.lang.String)
 	 */
 	@Override
-	public Relationship voidRelationship(Relationship relationship, String voidReason) throws APIException {
-		if (relationship.getVoided()) {
-			return relationship;
-		}
-		
-		relationship.setVoided(true);
-		if (relationship.getVoidedBy() == null) {
-			relationship.setVoidedBy(Context.getAuthenticatedUser());
-		}
-		if (voidReason != null) {
-			relationship.setVoidReason(voidReason);
+	public Relationship voidRelationship(@NonNull final Relationship relationship, String voidReason) throws APIException {
+		if (relationship ==null) {
+			return null;
 		}
-		relationship.setDateVoided(new Date());
-		
+			
 		return Context.getPersonService().saveRelationship(relationship);
 	}
+
 	
 	/**
 	 * @see org.openmrs.api.PersonService#unvoidRelationship(org.openmrs.Relationship)
 	 */
 	@Override
-	public Relationship unvoidRelationship(Relationship relationship) throws APIException {
+	public Relationship unvoidRelationship(Relationship relationship) throws APIException {app://resources/notifications.html#
```

Comment:
Please don't include this change in this PR.

```suggestion
	public Relationship unvoidRelationship(Relationship relationship) throws APIException {
```